### PR TITLE
Add agent supported features list needed by Extensions 

### PR DESCRIPTION
**Pull request by larohra**
## Description
This PR adds the ability for the agent to pass the list of supported features needed by extensions (like ETP). 
It also does a bit of refactoring around the code.

Issue # 

---

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made

### Quality of Code and Contribution Guidelines
- [ ] I have read the [contribution guidelines](https://github.com/Azure/WALinuxAgent/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: azurelinuxagent/common/agent_supported_feature.py [hunk 1]
```diff
@@ -58,15 +59,35 @@ def __init__(self):
                                                   supported=_MultiConfigFeature.__SUPPORTED)
 
 
+class _ETPFeature(AgentSupportedFeature):
```
Comment (larohra) [inline azurelinuxagent/common/agent_supported_feature.py:62]: Keeping in tune with the Supported Features, added a new feature for ETP and using this everywhere to check if it is supported or not

File: azurelinuxagent/ga/send_telemetry_events.py [hunk 2]
```diff
@@ -63,7 +62,7 @@ def get_thread_name():
         return SendTelemetryEventsHandler._THREAD_NAME
 
     def run(self):
-        logger.info("Start Extension Telemetry service.")
+        logger.info("Start SendTelemetryHandler service.")
```
Comment (larohra) [inline azurelinuxagent/ga/send_telemetry_events.py:65]: While debugging some other issue found this bad log so fixed it here

File: azurelinuxagent/common/agent_supported_feature.py [hunk 3]
```diff
@@ -58,15 +59,35 @@ def __init__(self):
...
+                                          supported=self.__SUPPORTED)
+
+
 __CRP_SUPPORTED_FEATURES = {
-    CRPSupportedFeatureNames.MultiConfig: _MultiConfigFeature()
+    SupportedFeatureNames.MultiConfig: _MultiConfigFeature()
+}
+
+
+__EXTENSION_SUPPORTED_FEATURES = {
```
Comment (narrieta) [inline azurelinuxagent/common/agent_supported_feature.py:79]: I would just add a comment in these 2 lists saying the first/second one is the list of features that the agent supports and we advertise to CRP/Extensions.

From the variable names it reads like it is CRP/Extensions who support the features.

**Review by narrieta — APPROVED** [review]
Minor suggestion, but LGTM

File: azurelinuxagent/common/agent_supported_feature.py (hunk 3 above)
Comment (larohra) [reply azurelinuxagent/common/agent_supported_feature.py]: Done

**Review by narrieta — APPROVED** [review]

**Pull request merged**